ocean_lib/web3_internal/wallet.py

```diff
@@ -123,20 +123,26 @@ def sign_tx(
             )
         else:
             nonce = Wallet._get_nonce(self.web3, account.address)
+            from ocean_lib.web3_internal.contract_base import ContractBase
+
+            if not gas_price:
+                fee_tx = ContractBase.get_max_fee_per_gas(web3=self.web3, tx=tx)
+                if fee_tx:
```

Comment:
this still seems duplicated to what I'm seeing in contract_base. Could we improve it even more?

Reply:
Here I just call the function, but there are different fields to fill or different conditions.

Reply:
I think it will become more clear after you refactor the handling in contract base and moving the function separately. You can handle this comment last.